workers/exclusive-content/index.js

```diff
@@ -0,0 +1,164 @@
+/*
+ * Secret MASTERKEY, stored in CF worker settings
+ * @param {string}  MASTERKEY
+ */
+
+const headers = {
+  'Access-Control-Allow-Origin': '*',
+  'Access-Control-Allow-Methods': 'POST,OPTIONS',
+  'Access-Control-Allow-Headers': 'Content-Type',
+  'Access-Control-Max-Age': '86400'
+}
+
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest(event.request))
+})
+
+async function handleRequest(request) {
+  if (request.method === 'POST') {
+    const headers = new Map(request.headers)
+    if (headers.get('content-type') !== 'application/json') {
+      return new Response('Unsupported content type. Use application/json', {
+        headers,
+        status: 415
+      })
+    }
+    const url = new URL(request.url)
+    const path = url.pathname
+    if ((path !== '/deriveNewKey') & (path !== '/deriveKey')) {
+      return new Response('Not found', { headers, status: 404 })
+    }
+    const body = await request.json()
+    if (path === '/deriveNewKey') {
+      if (Object.keys(body).includes('paymentPointer')) {
+        return handleNewKeyDerivation(body)
+      } else {
+        return new Response('Payment pointer not part of request body', {
+          headers,
+          status: 400
+        })
+      }
+    }
+    if (path === '/deriveKey') {
+      if (
+        Object.keys(body).includes('paymentPointer') &&
+        Object.keys(body).includes('nonce') &&
+        Object.keys(body).includes('encVerifier') &&
+        Object.keys(body).includes('initVector') &&
+        Object.keys(body).includes('receipt')
+      ) {
+        return handleExKeyDerivation(body)
+      } else {
+        return new Response('Input variables missing', {
+          headers,
+          status: 400
+        })
+      }
+    }
+  } else if (request.method === 'OPTIONS') {
+    return new Response('OK', { headers })
+  } else {
+    return new Response('Not found', { headers, status: 404 })
+  }
+}
+
+async function handleNewKeyDerivation(body) {
+  const nonce = getRandomValue()
+  const keyBuffer = await deriveKey(encode(body.paymentPointer + nonce))
```

Comment:
making the deriveKey method just take 2 arguments (paymentPointer, nonce) and do the concatenation + encoding inside could save a little duplication